Implement white-space-collapse & text-wrap as longhands of white-space

RWDavid opened the pull request:
#### eff0d22edb04a7bc761bc1fb0428abef8ccbb1ba
<pre>
Implement white-space-collapse &amp; text-wrap as longhands of white-space
<a href="https://bugs.webkit.org/show_bug.cgi?id=256927">https://bugs.webkit.org/show_bug.cgi?id=256927</a>
rdar://109486221

Reviewed by Tim Nguyen.

This commit lays the groundwork for future work on white-space-collapse
and text-wrap. White-space is now parsed as a shorthand property, and
the longhand values for white-space (white-space-collapse and text-wrap)
are set accordingly.

The whiteSpace value in m_inheritedFlags in RenderStyle has been removed.
The RenderStyle.whiteSpace() method still exists, but it uses the two
longhand variables above to reconstruct the original value of whiteSpace.
The setWhiteSpace method has been removed, and any existing references
to setWhiteSpace have been replaced with the corresponding calls to
setWhiteSpaceCollapse and setTextWrap.

Further work needs to be done to properly integrate the white-space-collapse
and text-wrap longhands into WebKit. Currently, this commit preserves
the original behavior of white-space processing, but it assumes that
white-space-collapse and text-wrap only take on the combinations of
values that are achievable via the shorthand values of white-space.
(Otherwise, the whiteSpace getter in RenderStyle cannot reconstruct the
&apos;original&apos; value of white-space.) Thus, the codebase must replace
existing references to RenderStyle.whiteSpace() with appropriate references
to RenderStyle.whiteSpaceCollapse() and RenderStyle.textWrap() before
individually modifying the white-space-collapse and text-wrap properties.

* LayoutTests/fast/css/getComputedStyle/computed-style-expected.txt:
* LayoutTests/fast/css/getComputedStyle/computed-style-without-renderer-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-cascade/all-prop-revert-layer-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-cascade/all-prop-revert-layer.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/parsing/white-space-shorthand-text-wrap-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/white-space-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/white-space.html:
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/ipad/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/css/css-cascade/all-prop-revert-layer-expected.txt:
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/svg/css/getComputedStyle-basic-expected.txt:
* Source/WebCore/animation/CSSPropertyAnimation.cpp:
(WebCore::CSSPropertyAnimationWrapperMap::CSSPropertyAnimationWrapperMap):
* Source/WebCore/css/CSSComputedStyleDeclaration.cpp:
(WebCore::CSSComputedStyleDeclaration::getPropertyCSSValue const):
* Source/WebCore/css/CSSPrimitiveValueMappings.h:
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/ComputedStyleExtractor.cpp:
(WebCore::ComputedStyleExtractor::whiteSpaceShorthandValue):
(WebCore::ComputedStyleExtractor::propertyValue):
(WebCore::ComputedStyleExtractor::valueForPropertyInStyle):
* Source/WebCore/css/ComputedStyleExtractor.h:
* Source/WebCore/css/ShorthandSerializer.cpp:
(WebCore::ShorthandSerializer::serialize):
(WebCore::ShorthandSerializer::serializeWhiteSpace const):
* Source/WebCore/css/parser/CSSParserContext.cpp:
(WebCore::CSSParserContext::CSSParserContext):
* Source/WebCore/css/parser/CSSPropertyParser.cpp:
(WebCore::initialValueForLonghand):
(WebCore::CSSPropertyParser::consumeWhiteSpaceShorthand):
(WebCore::CSSPropertyParser::parseShorthand):
* Source/WebCore/css/parser/CSSPropertyParser.h:
* Source/WebCore/editing/EditingStyle.cpp:
(WebCore::EditingStyle::conflictsWithInlineStyleOfElement const):
(WebCore::EditingStyle::removeStyleFromRulesAndContext):
(WebCore::StyleChange::StyleChange):
* Source/WebCore/html/HTMLInputElement.cpp:
(WebCore::HTMLInputElement::createInnerTextStyle):
* Source/WebCore/html/HTMLPreElement.cpp:
(WebCore::HTMLPreElement::collectPresentationalHintsForAttribute):
* Source/WebCore/html/HTMLTableCellElement.cpp:
(WebCore::HTMLTableCellElement::collectPresentationalHintsForAttribute):
* Source/WebCore/html/HTMLTextAreaElement.cpp:
(WebCore::HTMLTextAreaElement::collectPresentationalHintsForAttribute):
* Source/WebCore/html/track/TextTrackCueGeneric.cpp:
(WebCore::TextTrackCueGenericBoxElement::applyCSSProperties):
* Source/WebCore/html/track/VTTCue.cpp:
(WebCore::VTTCueBox::applyCSSProperties):
* Source/WebCore/rendering/RenderListItem.cpp:
(WebCore::RenderListItem::computeMarkerStyle const):
* Source/WebCore/rendering/RenderTheme.cpp:
(WebCore::RenderTheme::adjustStyle):
* Source/WebCore/rendering/RenderThemeMac.mm:
(WebCore::RenderThemeMac::adjustMenuListStyle const):
* Source/WebCore/rendering/style/RenderStyle.cpp:
(WebCore::RenderStyle::RenderStyle):
(WebCore::RenderStyle::changeRequiresLayout const):
(WebCore::RenderStyle::whiteSpace const):
* Source/WebCore/rendering/style/RenderStyle.h:
(WebCore::RenderStyle::whiteSpace const): Deleted.
(WebCore::RenderStyle::setWhiteSpace): Deleted.
* Source/WebCore/rendering/style/RenderStyleConstants.cpp:
* Source/WebCore/rendering/style/RenderStyleInlines.h:
(WebCore::RenderStyle::InheritedFlags::operator== const):
* Source/WebCore/style/PropertyAllowlist.cpp:
(WebCore::Style::isValidMarkerStyleProperty):
(WebCore::Style::isValidCueStyleProperty):
* Source/WebCore/style/StyleAdjuster.cpp:
(WebCore::Style::Adjuster::adjust const):
* Source/WebCore/svg/SVGTextContentElement.cpp:
(WebCore::SVGTextContentElement::collectPresentationalHintsForAttribute):

Canonical link: <a href="https://commits.webkit.org/265267@main">https://commits.webkit.org/265267@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/494fc6bb1da365d69c3578f7d8fb7d3650771986

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/10425 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/10655 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/10931 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/12065 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/9998 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/10437 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/12760 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/10608 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/12964 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/10584 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/11371 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/8780 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/12463 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/8675 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/9437 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/16697 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/9766 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/9589 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/12837 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/10021 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/8127 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/9179 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/9331 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2504 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/13434 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/9882 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/css/ComputedStyleExtractor.cpp:
```diff
@@ -574,6 +574,36 @@ static RefPtr<CSSValue> positionOffsetValue(const RenderStyle& style, CSSPropert
+Ref<CSSValue> ComputedStyleExtractor::getWhiteSpaceValue(const RenderStyle& style)
+{
+    WhiteSpaceCollapse wsc = style.whiteSpaceCollapse();
+    TextWrap tw = style.textWrap();
```
nit: no abbreviation, also you can use `auto` here.
```suggestion
    auto whiteSpaceCollapse = style.whiteSpaceCollapse();
    auto textWrap = style.textWrap();
```

nt1m commented on Source/WebCore/css/ShorthandSerializer.cpp:
```diff
@@ -1157,6 +1160,29 @@ String ShorthandSerializer::serializePageBreak() const
+String ShorthandSerializer::serializeWhiteSpace() const
+{
+    CSSValueID wsc = longhandValueID(0); // white-space-collapse
+    CSSValueID tw = longhandValueID(1); // text-wrap
```
ditto

nt1m commented on Source/WebCore/css/ShorthandSerializer.cpp:
```diff
@@ -1157,6 +1160,29 @@ String ShorthandSerializer::serializePageBreak() const
+    // Reachable only if white-space-collapse/text-wrap properties were individually set
+    // instead of being changed via white-space shorthand keyword. Fall back to separate
+    // longhand serializations for now.
+    return serializeLonghands();
```
```suggestion
    return emptyString();
```
This case happens only when you hit a combination that's unsupported by the `white-space` single value syntax, and since we don't implement the `<'white-space-collapse'> || <'text-wrap'> || <'white-space-trim'>` syntax atm, I think serializing to an empty string is a better option. It's what we usually do when the shorthand can't represent the longhands.

nt1m commented on Source/WebCore/rendering/style/RenderStyle.cpp:
```diff
@@ -2108,6 +2110,31 @@ int RenderStyle::computeLineHeight(const Length& lineHeightLength) const
+WhiteSpace RenderStyle::whiteSpace() const
+{
+    WhiteSpaceCollapse wsc = static_cast<WhiteSpaceCollapse>(m_inheritedFlags.whiteSpaceCollapse);
+    TextWrap tw = static_cast<TextWrap>(m_inheritedFlags.textWrap);
```
ditto

nt1m commented on Source/WebCore/css/parser/CSSPropertyParser.cpp:
```diff
@@ -2546,6 +2546,44 @@ bool CSSPropertyParser::consumeListStyleShorthand(bool important)
+    // FIXME: accept values other than the shorthand keywords
+    // FIXME: set whiteSpaceTrim values
+    RefPtr<CSSValue> whiteSpaceCollapseValue;
+    RefPtr<CSSValue> textWrapValue;
```
nit: I think we can omit "Value" here.

nt1m commented on Source/WebCore/css/parser/CSSPropertyParser.cpp:
```diff
@@ -2546,6 +2546,44 @@ bool CSSPropertyParser::consumeListStyleShorthand(bool important)
+    auto ident = consumeIdentRaw(m_range);
+    if (!ident)
+        return false;
+    if (!m_range.atEnd())
+        return false;
```
This can be removed, this check is done elsewhere.

RWDavid replied:
It seems like leaving out this check causes `imported/w3c/web-platform-tests/css/css-text/parsing/white-space-invalid.html` to fail.

mdubet commented on Source/WebCore/rendering/RenderThemeMac.mm:
```diff
@@ -1093,7 +1093,8 @@ static FloatRect inflateRect(const FloatRect& rect, const IntSize& size, const i
     style.setHeight(Length(LengthType::Auto));
 
     // White-space is locked to pre
```
This comment needs an update (or a removal altogether)

nt1m commented on Source/WebCore/css/ComputedStyleExtractor.cpp:
```diff
@@ -574,6 +574,36 @@ static RefPtr<CSSValue> positionOffsetValue(const RenderStyle& style, CSSPropert
 
+// FIXME: Replace calls to this function with `getCSSPropertyValuesForShorthandProperties`
+// once white-space longhands are completely implemented.
+Ref<CSSValue> ComputedStyleExtractor::getWhiteSpaceValue(const RenderStyle& style)
```
```suggestion
Ref<CSSValue> ComputedStyleExtractor::whiteSpacePropertyValue(const RenderStyle& style)
```

WebKit style guide says to not start function names with `get`.

nt1m commented on Source/WebCore/css/ComputedStyleExtractor.cpp:
```diff
@@ -574,6 +574,36 @@ static RefPtr<CSSValue> positionOffsetValue(const RenderStyle& style, CSSPropert
 }
 
+// FIXME: Replace calls to this function with `getCSSPropertyValuesForShorthandProperties`
+// once white-space longhands are completely implemented.
```
I don't think we can completely replace this with `getCSSPropertyValuesForShorthandProperties` fwiw, we still want to provide the shortest serialization, which `getCSSPropertyValuesForShorthandProperties` won't necessarily provide.

I'd just remove this FIXME.

nt1m commented on Source/WebCore/css/ComputedStyleExtractor.cpp:
```diff
@@ -574,6 +574,36 @@ static RefPtr<CSSValue> positionOffsetValue(const RenderStyle& style, CSSPropert
+    // feature flags for white-space are enabled, then fall back to separate longhands.
+    auto& settings = m_element->document().settings();
+    if (settings.cssWhiteSpaceCollapseEnabled() && settings.cssTextWrapEnabled())
+        return getCSSPropertyValuesForShorthandProperties(whiteSpaceShorthand());
```
Since we don't implement the `<'white-space-collapse'> || <'text-wrap'>` syntax, this is incorrect. I'd just remove this code for now.

nt1m commented on Source/WebCore/css/ComputedStyleExtractor.cpp:
```diff
@@ -574,6 +574,36 @@ static RefPtr<CSSValue> positionOffsetValue(const RenderStyle& style, CSSPropert
+        return getCSSPropertyValuesForShorthandProperties(whiteSpaceShorthand());
+
+    // The white-space property cannot be properly expressed in terms of its longhands
+    return CSSPrimitiveValue::create(CSSValueInvalid);
```
I would `return nullptr;` in this case (and have this function return a `RefPtr`)

nt1m commented on Source/WebCore/css/parser/CSSPropertyParser.cpp:
```diff
@@ -2546,6 +2547,44 @@ bool CSSPropertyParser::consumeListStyleShorthand(bool important)
+        whiteSpaceCollapse = CSSPrimitiveValue::create(CSSValuePreserve);
+        textWrap = CSSPrimitiveValue::create(CSSValueWrap);
+    } else
+        return false;
```
Replace this with a `switch` ?

nt1m commented on Source/WebCore/css/parser/CSSPropertyParser.cpp:
```diff
@@ -2546,6 +2547,44 @@ bool CSSPropertyParser::consumeListStyleShorthand(bool important)
+        return false;
+
+    // FIXME: accept values other than the shorthand keywords
+    // FIXME: set whiteSpaceTrim values
```
```suggestion
    // FIXME: Implement `<'white-space-collapse'> || <'text-wrap'>` syntax.
```

I'd remove the whiteSpaceTrim fixme for now, since I'd not specific to this function.

nt1m commented on Source/WebCore/editing/EditingStyle.cpp:
```diff
@@ -1430,6 +1431,21 @@ void EditingStyle::removeStyleFromRulesAndContext(StyledElement& element, Node*
+        // If white-space has changed, do not remove white-space longhand values.
+        // They are necessary for reconstructing the corresponding white-space shorthand value.
+        auto whiteSpaceCollapse = m_mutableStyle->getPropertyCSSValue(CSSPropertyWhiteSpaceCollapse);
+        auto whiteSpaceCollapseContext = computedStyle->m_mutableStyle->getPropertyCSSValue(CSSPropertyWhiteSpaceCollapse);
```
previousWhiteSpaceCollapse?

RWDavid replied:
I feel like "previous" might not be the most suitable term here. Basically, `whiteSpaceCollapseContext` is the style of the parent element while `whiteSpaceCollapse` is the style of the child element. Maybe something like `localWhiteSpaceCollapse`?

nt1m commented on Source/WebCore/editing/EditingStyle.cpp:
```diff
@@ -1430,6 +1431,21 @@ void EditingStyle::removeStyleFromRulesAndContext(StyledElement& element, Node*
+        bool whiteSpaceCollapseChanged = whiteSpaceCollapse != whiteSpaceCollapseContext;
+
+        auto textWrap = m_mutableStyle->getPropertyCSSValue(CSSPropertyTextWrap);
+        auto textWrapContext = computedStyle->m_mutableStyle->getPropertyCSSValue(CSSPropertyTextWrap);
```
previousTextWrap ?

nt1m commented on Source/WebCore/html/HTMLPreElement.cpp:
```diff
@@ -54,8 +54,10 @@ bool HTMLPreElement::hasPresentationalHintsForAttribute(const QualifiedName& nam
+        style.setProperty(CSSPropertyWhiteSpaceCollapse, CSSValuePreserve);
+        style.setProperty(CSSPropertyTextWrap, CSSValueWrap);
+    }
     else
```
The else goes on the previous line

```suggestion
    } else
```

nt1m commented on Source/WebCore/rendering/style/RenderStyle.cpp:
```diff
@@ -2108,6 +2110,31 @@ int RenderStyle::computeLineHeight(const Length& lineHeightLength) const
+
+    // Reachable only if white-space-collapse/text-wrap properties were individually set
+    // instead of being changed via white-space shorthand keyword. Do nothing for now
+    // since this is a temporary function.
```
// Reachable for combinations that can't be represented with the white-space syntax.

would be more accurate.

nt1m commented on Source/WebCore/css/ComputedStyleExtractor.h:
```diff
@@ -102,6 +102,7 @@ class ComputedStyleExtractor {
     Ref<CSSValue> getMaskShorthandValue();
     Ref<CSSValueList> getCSSPropertyValuesForGridShorthand(const StylePropertyShorthand&);
     Ref<CSSValue> fontVariantShorthandValue();
+    RefPtr<CSSValue> whiteSpacePropertyValue(const RenderStyle&);
```
whiteSpaceShorthandValue would be a better name given the other names here (sorry only noticed this now).

nt1m commented on Source/WebCore/css/ComputedStyleExtractor.cpp:
```diff
@@ -2866,11 +2887,6 @@ RefPtr<CSSValue> ComputedStyleExtractor::propertyValue(CSSPropertyID propertyID,
-        // Exit quickly, and avoid us ever having to update layout in this case.
-        return nullptr;
-    }
-
```
We need to keep this, but only skip this step when `propertyValue` is called from `copyProperties` (use a boolean argument with a default value so you only need to change one place)

nt1m commented on Source/WebCore/css/ComputedStyleExtractor.cpp:
```diff
@@ -2866,7 +2887,10 @@ RefPtr<CSSValue> ComputedStyleExtractor::propertyValue(CSSPropertyID propertyID,
-    if (!isExposed(propertyID, m_element->document().settings())) {
+    // FIXME: For now, we always allow access to white-space longhands.
+    // We should remove this hack once white-space longhands are no longer under a feature flag.
+    bool isWhiteSpaceLonghand = (propertyID == CSSPropertyWhiteSpaceCollapse || propertyID == CSSPropertyTextWrap);
+    if (!isWhiteSpaceLonghand && !isExposed(propertyID, m_element->document().settings())) {
```
Can we use shorthandForLonghand() + isExposed() like I suggested? instead of special casing white-space-collapse & text-wrap

RWDavid committed:
Implement white-space-collapse & text-wrap as longhands of white-space
https://bugs.webkit.org/show_bug.cgi?id=256927
rdar://109486221

Reviewed by Tim Nguyen.

This commit lays the groundwork for future work on white-space-collapse
and text-wrap. White-space is now parsed as a shorthand property, and
the longhand values for white-space (white-space-collapse and text-wrap)
are set accordingly.

The whiteSpace value in m_inheritedFlags in RenderStyle has been removed.
The RenderStyle.whiteSpace() method still exists, but it uses the two
longhand variables above to reconstruct the original value of whiteSpace.
The setWhiteSpace method has been removed, and any existing references
to setWhiteSpace have been replaced with the corresponding calls to
setWhiteSpaceCollapse and setTextWrap.

Further work needs to be done to properly integrate the white-space-collapse
and text-wrap longhands into WebKit. Currently, this commit preserves
the original behavior of white-space processing, but it assumes that
white-space-collapse and text-wrap only take on the combinations of
values that are achievable via the shorthand values of white-space.
(Otherwise, the whiteSpace getter in RenderStyle cannot reconstruct the
'original' value of white-space.) Thus, the codebase must replace
existing references to RenderStyle.whiteSpace() with appropriate references
to RenderStyle.whiteSpaceCollapse() and RenderStyle.textWrap() before
individually modifying the white-space-collapse and text-wrap properties.

* LayoutTests/fast/css/getComputedStyle/computed-style-expected.txt:
* LayoutTests/fast/css/getComputedStyle/computed-style-without-renderer-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-cascade/all-prop-revert-layer-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-cascade/all-prop-revert-layer.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/parsing/white-space-shorthand-text-wrap-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/white-space-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/white-space.html:
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/ipad/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/css/css-cascade/all-prop-revert-layer-expected.txt:
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/svg/css/getComputedStyle-basic-expected.txt:
* Source/WebCore/animation/CSSPropertyAnimation.cpp:
(WebCore::CSSPropertyAnimationWrapperMap::CSSPropertyAnimationWrapperMap):
* Source/WebCore/css/CSSComputedStyleDeclaration.cpp:
(WebCore::CSSComputedStyleDeclaration::getPropertyCSSValue const):
* Source/WebCore/css/CSSPrimitiveValueMappings.h:
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/ComputedStyleExtractor.cpp:
(WebCore::ComputedStyleExtractor::whiteSpaceShorthandValue):
(WebCore::ComputedStyleExtractor::propertyValue):
(WebCore::ComputedStyleExtractor::valueForPropertyInStyle):
* Source/WebCore/css/ComputedStyleExtractor.h:
* Source/WebCore/css/ShorthandSerializer.cpp:
(WebCore::ShorthandSerializer::serialize):
(WebCore::ShorthandSerializer::serializeWhiteSpace const):
* Source/WebCore/css/parser/CSSParserContext.cpp:
(WebCore::CSSParserContext::CSSParserContext):
* Source/WebCore/css/parser/CSSPropertyParser.cpp:
(WebCore::initialValueForLonghand):
(WebCore::CSSPropertyParser::consumeWhiteSpaceShorthand):
(WebCore::CSSPropertyParser::parseShorthand):
* Source/WebCore/css/parser/CSSPropertyParser.h:
* Source/WebCore/editing/EditingStyle.cpp:
(WebCore::EditingStyle::conflictsWithInlineStyleOfElement const):
(WebCore::EditingStyle::removeStyleFromRulesAndContext):
(WebCore::StyleChange::StyleChange):
* Source/WebCore/html/HTMLInputElement.cpp:
(WebCore::HTMLInputElement::createInnerTextStyle):
* Source/WebCore/html/HTMLPreElement.cpp:
(WebCore::HTMLPreElement::collectPresentationalHintsForAttribute):
* Source/WebCore/html/HTMLTableCellElement.cpp:
(WebCore::HTMLTableCellElement::collectPresentationalHintsForAttribute):
* Source/WebCore/html/HTMLTextAreaElement.cpp:
(WebCore::HTMLTextAreaElement::collectPresentationalHintsForAttribute):
* Source/WebCore/html/track/TextTrackCueGeneric.cpp:
(WebCore::TextTrackCueGenericBoxElement::applyCSSProperties):
* Source/WebCore/html/track/VTTCue.cpp:
(WebCore::VTTCueBox::applyCSSProperties):
* Source/WebCore/rendering/RenderListItem.cpp:
(WebCore::RenderListItem::computeMarkerStyle const):
* Source/WebCore/rendering/RenderTheme.cpp:
(WebCore::RenderTheme::adjustStyle):
* Source/WebCore/rendering/RenderThemeMac.mm:
(WebCore::RenderThemeMac::adjustMenuListStyle const):
* Source/WebCore/rendering/style/RenderStyle.cpp:
(WebCore::RenderStyle::RenderStyle):
(WebCore::RenderStyle::changeRequiresLayout const):
(WebCore::RenderStyle::whiteSpace const):
* Source/WebCore/rendering/style/RenderStyle.h:
(WebCore::RenderStyle::whiteSpace const): Deleted.
(WebCore::RenderStyle::setWhiteSpace): Deleted.
* Source/WebCore/rendering/style/RenderStyleConstants.cpp:
* Source/WebCore/rendering/style/RenderStyleInlines.h:
(WebCore::RenderStyle::InheritedFlags::operator== const):
* Source/WebCore/style/PropertyAllowlist.cpp:
(WebCore::Style::isValidMarkerStyleProperty):
(WebCore::Style::isValidCueStyleProperty):
* Source/WebCore/style/StyleAdjuster.cpp:
(WebCore::Style::Adjuster::adjust const):
* Source/WebCore/svg/SVGTextContentElement.cpp:
(WebCore::SVGTextContentElement::collectPresentationalHintsForAttribute):

Canonical link: https://commits.webkit.org/265267@main
